service/controller/chartconfig/v7/resource/chart/current.go

```diff
@@ -17,7 +18,11 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 		return nil, microerror.Mask(err)
 	}
 
-	if key.IsCordoned(customObject) {
+	// Delete annotation is set when the chartconfig CR has been migrated to an
+	// app CR and can be safely deleted.
+	_, hasDeleteAnnotation := customObject.Annotations[annotation.DeleteCustomResourceOnly]
+
+	if key.IsCordoned(customObject) && !hasDeleteAnnotation {
```

Comment:
I don't get this part, why we don't simply use it `Cordon` annotation? 
I also look into `service/controller/chartconfig/v7/resource/chart/delete.go` but it only return nil when it has`deleteAnnotation`. Perhaps you want to keep update & create CRUD features as on?

Reply:
We cordon the app at the start of the migration. Once the app status is deployed we add this annotation.

This indicates the migration is complete and we delete the chartconfig CR. Here is the code in cluster-operator.

https://github.com/giantswarm/cluster-operator/blob/f9ca993418f053d705dade4d2cf72976cee40f4d/pkg/v20/resource/appmigration/create.go#L117-L138